plugin-dev/Source/Sentry/Private/Apple/AppleSentrySubsystem.cpp

```diff
@@ -34,10 +37,12 @@
 #include "Misc/Paths.h"
 #include "UObject/GarbageCollection.h"
 #include "UObject/UObjectThreadContext.h"
-#include "Utils/SentryLogUtils.h"
 
 void FAppleSentrySubsystem::InitWithSettings(const USentrySettings* settings, USentryBeforeSendHandler* beforeSendHandler, USentryBeforeBreadcrumbHandler* beforeBreadcrumbHandler, USentryTraceSampler* traceSampler)
 {
+	isScreenshotAttachmentEnabled = settings->AttachScreenshot;
+	isGameLogAttachmentEnabled = settings->EnableAutoLogAttachment;
```

Comment:
Should these go on the actual `options`? We're starting to mix `settings.` with `options.` here. I'd really prefer the `options` to be the source of truth of what options exist and what options the SDK.

Reply:
These two flags are cached separately here since they don't necessarily have a direct representation in Cocoa SDK options:

- `isGameLogAttachmentEnabled` is Unreal-specific feature
- `isScreenshotAttachmentEnabled` can't be just set via `options.attachScreenshot` as it's an iOS-only thing in `sentry-coca`

Accessing them using `FSentryModule::Get().GetSettings()` could cause unwanted ensures when called from a non-game thread.

Reply:
```
attachScreenshot
```

This option (with this name) also exists on Android, .NET, Godot, Unity.

Reply:
I might be missing the concern here. These flags are passed to the Cocoa SDK’s `options` within the `startWithConfigureOptions` block right after we make copies of them here (mainly for convenience as we use those in a few other places beyond initialization). Or is this about the naming?

Reply:
Yes, it's a common naming convention for Unreal - [ISettingsModule::RegisterSettings](https://dev.epicgames.com/documentation/en-us/unreal-engine/API/Developer/Settings/ISettingsModule/RegisterSettings/1)